tests/e2e/stepDefinitions/slideViewerContext.js

```diff
@@ -0,0 +1,29 @@
+const { Given, When, Then } = require('@cucumber/cucumber')
+const { expect } = require('@playwright/test')
+const { SlideViewerPage } = require('../pageObjects/slideViewerPage')
+const { createFile } = require('../utils/helpers')
+const { config } = require('../cucumber.conf')
+const slideViewerPage = new SlideViewerPage()
+
+Given('user {string} has logged in', async function (user) {
+  await slideViewerPage.navigate()
+  await slideViewerPage.login(config.adminUser, config.adminPassword)
+  await expect(page.locator(slideViewerPage.filesContainerSelector)).toBeVisible()
+})
+
+Given('user has uploaded the markdown file {string}', async function (fileName) {
+  await createFile(fileName)
+  await page.reload()
+})
+
+When(
+  'user previews a markdown file {string} in presentation viewer mode',
+  async function (fileName) {
+    await slideViewerPage.openMDFileInPresentationViewer()
+    await expect(page.locator(slideViewerPage.slideViewerHomepageSelector)).toBeVisible()
+  }
+)
+
+Then('markdown content should be rendered as slides in a presentation format', async function () {
+  await expect(page.locator(slideViewerPage.slidesContainerSelector)).toBeVisible()
```

Comment:
that is a very basic check, but I think that is fine for now and more specific checks can be added later